### DISCUSSION TITLE
Share groups of panels between dashboards

### DISCUSSION THREAD

**Discussion by jameswestover** [grafana/grafana#38347] [Dashboards, Panels & Visualizations]
When we deploy applications we issue a default dashboard to our users that gives them basic metrics about their applications (cpu / mem usage, tps, latencies, etc). We've found over time we want to change that base template (enhance it with new metrics, remove or update bad queries) it becomes massively cumbersome to get thousands of dashboards updated or regenerated.

If we had the ability to layer dashboards, such that when a top layered dashboards was page refreshed, it would get the latest base dashboard, put those panels on the upper layer dashboard in a read-only mode, then layer in all current panels, we would be able to prevent this constant churn of dashboard creation.

In a large organization of users where the same data is useful in different contexts, this would allow us to reduce the problem of "we updated a panel over here, but the same panel on a different dashboard didn't get updated and now its reporting the wrong data". We could have a single dashboard showing that base data, that effectively gets imported into any number of other dashboards, such that when the imported dashboard is updated all linked dashboards also get those updates.

**Comment by torkelo** [comment]
related to this feature request: https://github.com/grafana/grafana/issues/1803 but more for groups of panels

**Comment by marefr** [comment]
@jameswestover I think what you're referring to may be a good use case for programmatic generation of dashboard using for example https://github.com/grafana/grafonnet-lib.

**Comment by jameswestover** [comment]
Generating the dashboard isn't the problem. We have a programatic way to do that. The problem is when 2000 people have programmatically created a templated dashboard, how do you update them? How do you update subsets of panels on a generated dashboard for all dashboards that were generated by a template.

**Comment by marefr** [comment]
The changes would need to happen programmatically as well. Then re-generate all dashboards and re-provision them.

**Comment by asafm** [comment]
I want to lay out my thinking on this matter:
1. Say Grafana had the ability to add a row (group of panels) from an existing dashboard, so it's like an "include" of a row. This opens up the possibility to re-use. 
2. When you write a microservice, 99% you are either using your organization internal libraries or using an open source library, or both. 
3. Given (1), assume you have existing dashboards (say they contain 1 row) for each sub library - they can be community maintained some how (the open source libraries ones) and internally maintained for the organization libraries.
4. When a team deploys a new microservice, it's becomes a matter of lego blocks to compose your own service dashboard. It also allows you to click and go to the relevant dashboard which may contain more detailed panels (other rows)

**Reply by saminzadeh** [reply]
This is how I view it as well. 

I work in an engineering org of over 3k+ engineers. We have Platform teams that create dashboards (powered by template variables) that are used by Product teams. Ideally, Product engineers can embed an instance of these panels in their own dashboards with the correct template variables.

I believe a large part of the challenge is going to be sharing templates variables in those Library Panels between dashboards / panels.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
